src/IO/Compare/Compare.cxx

```diff
@@ -91,72 +91,142 @@ int Checkerboard(itk::wasm::Pipeline &pipeline, const TMovingImage *movingImage,
+      }
+      filter->SetCheckerPattern(checkerPattern);
+    }
 
-  using SplitterType = itk::ImageRegionSplitterSlowDimension;
-  auto splitter = SplitterType::New();
-  const unsigned int numberOfSplits = splitter->GetNumberOfSplits(largestRegion, maxTotalSplits);
+    // Split handling
+    using ROIFilterType = itk::ExtractImageFilter<FixedImageType, FixedImageType>;
+    filter->UpdateOutputInformation();
+    using RegionType = typename FixedImageType::RegionType;
+    const RegionType largestRegion(filter->GetOutput()->GetLargestPossibleRegion());
 
-  if (split >= numberOfSplits)
-  {
-    std::cerr << "Error: requested split: " << split << " is outside the number of splits: " << numberOfSplits << std::endl;
-    return EXIT_FAILURE;
-  }
+    using SplitterType = itk::ImageRegionSplitterSlowDimension;
+    auto splitter = SplitterType::New();
+    const unsigned int numberOfSplits = splitter->GetNumberOfSplits(largestRegion, maxTotalSplits);
 
-  if (!numberOfSplitsStreamOption->empty())
-  {
-    numberOfSplitsStream.Get() << numberOfSplits;
-  }
+    if (split >= numberOfSplits)
+    {
+      std::cerr << "Error: requested split: " << split << " is outside the number of splits: " << numberOfSplits << std::endl;
+      return EXIT_FAILURE;
+    }
 
-  RegionType requestedRegion(largestRegion);
-  splitter->GetSplit(split, numberOfSplits, requestedRegion);
-  auto roiFilter = ROIFilterType::New();
-  roiFilter->SetExtractionRegion(requestedRegion);
-  roiFilter->SetInput(filter->GetOutput());
+    if (!numberOfSplitsStreamOption->empty())
+    {
+      numberOfSplitsStream.Get() << numberOfSplits;
+    }
 
-  try
-  {
-    roiFilter->Update();
+    RegionType requestedRegion(largestRegion);
+    splitter->GetSplit(split, numberOfSplits, requestedRegion);
+    auto roiFilter = ROIFilterType::New();
+    roiFilter->SetExtractionRegion(requestedRegion);
+    roiFilter->SetInput(filter->GetOutput());
+
+    try
+    {
+      roiFilter->Update();
+    }
+    catch (std::exception &error)
+    {
+      std::cerr << "Error: " << error.what() << std::endl;
+      return EXIT_FAILURE;
+    }
+
+    using OutputImageType = itk::wasm::OutputImage<FixedImageType>;
+    OutputImageType outputImage;
+    pipeline.add_option("output-image", outputImage, "Output image")->required()->type_name("OUTPUT_IMAGE");
+
+    ITK_WASM_PARSE(pipeline);
+
+    outputImage.Set(roiFilter->GetOutput());
+
+    return EXIT_SUCCESS;
   }
-  catch (std::exception &error)
+  else
   {
-    std::cerr << "Error: " << error.what() << std::endl;
-    return EXIT_FAILURE;
-  }
+    // cyan-magenta or blend
+
+    // using FilterType = itk::ComposeImageFilter<FixedImageType>;
+    // auto filter = FilterType::New();
+    // filter->SetInput(0, fixedImage);
+    // filter->SetInput(1, rescaleFilter->GetOutput());
+
+    // using PipelineOutputType = typename itk::VectorImage<typename FixedImageType::PixelType, FixedImageType::ImageDimension>;
```

Comment:
Perhaps we should also output a two-component VectorImage? With Checkboard, the second component is what `swapImageOrder` enables now? We could simplify to one pipeline, enable cyan/magenta with imageMix or grayscale blent with imageMix with or without checkerboard?

Reply:
Good idea!  Faster swap image order toggling.  

And you put your finger right on the last issue (in my mind) with this PR:  Only getting a 1 component image out in JS land with this ComposeImagerFilter...